### PR TITLE
Adding support for subsonic cover art download

### PR DESCRIPTION
**Pull request by davidgfnet**
Hey, just rebased and fixed some stuff, hope you guys are able to merge this.
Thanks for your work!

### CODE REVIEW

File: src/covers/albumcoverloader.h [hunk 1]
```diff
@@ -104,11 +105,12 @@ class AlbumCoverLoader : public QObject {
   QQueue<Task> tasks_;
   QMap<QNetworkReply*, Task> remote_tasks_;
   QMap<QString, Task> remote_spotify_tasks_;
+  QMap<QString, Task> remote_subsonic_tasks_;
   quint64 next_id_;
 
   NetworkAccessManager* network_;
 
-  bool connected_spotify_;
+  bool connected_spotify_, connected_subsonic_;
```
Comment (hatstand) [inline src/covers/albumcoverloader.h]: One per line

File: src/internet/subsonic/subsonicservice.cpp [hunk 2]
```diff
@@ -576,6 +586,24 @@ void SubsonicLibraryScanner::GetAlbumList(int offset) {
              SLOT(OnGetAlbumListFinished(QNetworkReply*, int)), reply, offset);
 }
 
+void SubsonicLibraryScanner::GetAlbumCover(const QString& id) {
+  QUrl url = service_->BuildRequestUrl("getCoverArt");
+  url.addQueryItem("id", id);
+  url.addQueryItem("size", "1024");
```
Comment (hatstand) [inline src/internet/subsonic/subsonicservice.cpp]: Add a constant for `1024`

File: src/internet/subsonic/subsonicservice.h [hunk 3]
```diff
@@ -101,6 +101,9 @@ class SubsonicService : public InternetService {
 
   LoginState login_state() const { return login_state_; }
 
+  Q_INVOKABLE void LoadImage(const QString& id);
+  void emitImageLoaded(const QString& id, const QImage& image);
```
Comment (hatstand) [inline src/internet/subsonic/subsonicservice.h]: `EmitImageLoaded`

**Comment by davidgfnet** [comment]
Just updated this patch, for some reason I wasn't getting notifications on this. Now should be fixed, haven't really tested though, since it's pretty straight forward.
Cheers!

**Pull request merged**